src/Semaphore.sol

```diff
@@ -70,42 +103,55 @@ contract Semaphore is ISemaphore, SemaphoreCore, Verifier, SemaphoreGroups {
         uint8[] calldata proofPathIndices
     ) public {
         if (msg.sender != manager) revert Unauthorized();
+        if (getDepth(groupId) == 0) revert InvalidId();
 
-        _removeMember(groupId, identityCommitment, proofSiblings, proofPathIndices);
+        groups[groupId].remove(identityCommitment, proofSiblings, proofPathIndices);
+
+        uint256 root = getRoot(groupId);
+        rootHistory[root] = RootHistory({
+            groupId: uint128(groupId),
+            timestamp: uint128(block.timestamp)
+        });
+
+        emit MemberRemoved(groupId, identityCommitment, groups[groupId].root);
     }
 
     ///////////////////////////////////////////////////////////////////////////////
     ///                          PROOF VALIDATION LOGIC                        ///
     //////////////////////////////////////////////////////////////////////////////
 
-    /// @notice Wether the zero-knowledge proof is valid.
+    /// @notice Reverts if the zero-knowledge proof is invalid.
     /// @param root The of the Merkle tree
     /// @param groupId The id of the Semaphore group
     /// @param signalHash A keccak256 hash of the Semaphore signal
     /// @param nullifierHash The nullifier hash
     /// @param externalNullifierHash A keccak256 hash of the external nullifier
     /// @param proof The zero-knowledge proof
-    /// @return Wether the proof is valid or not
     /// @dev  Note that a double-signaling check is not included here, and should be carried by the caller.
-    function isValidProof(
+    function verifyProof(
         uint256 root,
         uint256 groupId,
         uint256 signalHash,
         uint256 nullifierHash,
         uint256 externalNullifierHash,
         uint256[8] calldata proof
-    ) public view returns (bool) {
-        if (rootHistory[root] != groupId && getNumberOfLeaves(groupId) > 0) revert InvalidRoot();
+    ) public view {
+        RootHistory memory rootData = rootHistory[root];
+
+        if (
+            rootData.groupId != groupId &&
```

Comment:
Are you sure you want `&&` and not `||`?

Reply:
Oops, yeah, that should be `||`, updated (& updated tests to catch this too)